src/http.c

```diff
@@ -262,27 +262,64 @@ int response_headers_complete_cb(http_parser *p)
   return 0;
 }
 
-void gzip_init_headers(http_response_t *response)
+void http_init_response_headers(http_response_t *response, bool compresed)
 {
-  gzip_state_t *state = response->gzip_state;
-  strcat(state->http_header, response->status_line);
-  strcat(state->http_header, "\r\n");
+  memset(response->http_header, 0, sizeof response->http_header);
+  strcat(response->http_header, response->status_line);
+  strcat(response->http_header, "\r\n");
   for (int i = 0; i < response->num_headers; ++i)
   {
     // Skip responses of non compressed headers
-    if (strcasecmp(response->headers[i][0], "Content-Length") == 0 || strcasecmp(response->headers[i][0], "Accept-Ranges") == 0)
+    if (compresed && (strcasecmp(response->headers[i][0], "Content-Length") == 0 || strcasecmp(response->headers[i][0], "Accept-Ranges") == 0))
     {
       continue;
     }
 
-    strcat(state->http_header, response->headers[i][0]);
-    strcat(state->http_header, ": ");
-    strcat(state->http_header, response->headers[i][1]);
-    strcat(state->http_header, "\r\n");
+    strcat(response->http_header, response->headers[i][0]);
+    strcat(response->http_header, ": ");
+    strcat(response->http_header, response->headers[i][1]);
+    strcat(response->http_header, "\r\n");
   }
-  // Add gzip headers
-  strcat(state->http_header, "Transfer-Encoding: chunked\r\n");
-  strcat(state->http_header, "Content-Encoding: gzip\r\n");
+  if (compresed)
+  {
+    // Add gzip headers
+    strcat(response->http_header, "Transfer-Encoding: chunked\r\n");
+    strcat(response->http_header, "Content-Encoding: gzip\r\n");
+  }
+  strcat(response->http_header, "Via: bproxy\r\n");
+  strcat(response->http_header, "\r\n");
+  response->http_header_len = strlen(response->http_header);
+}
+
+void http_init_request_headers(http_link_context_t *context)
+{
+  http_request_t *request = &context->request;
+  memset(request->http_header, 0, sizeof request->http_header);
+  strcat(request->http_header, request->status_line);
+  strcat(request->http_header, "\r\n");
+  for (int i = 0; i < request->num_headers; ++i)
+  {
+    strcat(request->http_header, request->headers[i][0]);
+    strcat(request->http_header, ": ");
+    strcat(request->http_header, request->headers[i][1]);
+    strcat(request->http_header, "\r\n");
+  }
+  // Forwarded: by=<identifier>; for=<identifier>; host=<host>; proto=<http|https>
+  char forwarded_header[1024];
+  memset(forwarded_header, 0, sizeof forwarded_header);
+
+  char *proto;
+  if (context->https)
+  {
+    proto = "https";
+  }
+  else
+  {
+    proto = "http";
+  }
+  sprintf(forwarded_header, "Forwarded: by=%s; for=%s; host=%s; proto=%s\r\n", "bproxy", context->peer_ip, request->host, proto);
```

Comment:
I don't want to be too smart in this case, nevertheless PR looks very good... but isn't it there simpler way to add header with `http_parser`? Not sure in this case, correct me if I am wrong.